frictionless/portals/zenodo/manager.py

```diff
@@ -1,33 +1,176 @@
+            deposition_id = self.control.deposition_id
+            if not deposition_id:
+                # Create a deposition resource
+                deposition_id = client.create(
+                    token=self.control.apikey, base_url=self.control.base_url
+                )
+            metafn = Path(self.control.metafn).expanduser()
+            client.upload_meta(
+                token=self.control.apikey,
+                metafn=metafn,
+                depid=deposition_id,
+            )
+
+            # Process resources
+            resources = []
+            for key, resource in enumerate(package.resources):
+                if resource.data:
+                    resource_file_name = f"{resource.name}.json" or f"resource{key}.json"
+                    resource_path = os.path.join(
+                        self.control.tmp_path or "", resource_file_name
+                    )
+                    resource.to_json(resource_path)
+                    resources.append(Path(resource_path).expanduser())
+                    continue
+
+                resource_path = resource.path or ""
+                if resource_path.startswith(("http://", "https://")):
+                    continue
+
+                if resource.basepath:
+                    resource_path = os.path.join(
+                        str(resource.basepath), str(resource.path)
+                    )
+                resources.append(Path(resource_path).expanduser())
+            package_path = os.path.join(self.control.tmp_path or "", "datapackage.json")
+            package.to_json(package_path)
+
+            # Upload package and resources
+            client.upload_data(
+                token=self.control.apikey,
+                datafn=Path(package_path).expanduser(),
+                depid=deposition_id,
+                base_url=self.control.base_url,
+            )
+            for resource_path in resources:
+                resource_path = Path(resource_path).expanduser()
+                client.upload_data(
+                    token=self.control.apikey,
+                    datafn=resource_path,
+                    depid=deposition_id,
+                    base_url=self.control.base_url,
+                )
+            return deposition_id
+        except Exception as exception:
+            note = "Zenodo API error" + repr(exception)
+            raise FrictionlessException(note)
+
+
+def get_package(files: List, title: str, formats: List[str]) -> Package:
+    package = Package()
+    package.title = title
+    for file in files:
+        path = file["links"]["self"]
+        if path.endswith(("datapackage.json", "datapackage.yaml")):
+            return Package.from_descriptor(path, title=title)
+        if path.endswith(("zip")):
+            try:
+                package = Package.from_zip(path, title=title)
+            except FrictionlessException as exception:
+                if not "[Errno 2] No such file or directory" in str(exception):
```

Comment:
This error is raised when `from_zip` function doesn't find "datapackage.json" file after it unzips the folder and searches for 'datapackage.json' or 'datapackage.yaml' file in unzipped folder.

https://github.com/frictionlessdata/framework/blob/main/frictionless/package/package.py#L838
https://github.com/frictionlessdata/framework/blob/main/frictionless/package/package.py#L154

So this is to skip the error and continue reading files if it can't read the zipped folder. But I am not sure if this is the right way to skip the exception.

Need suggestion on it @aivuk @roll